gateway/src/apicast/policy/ip_check/client_ip.lua

```diff
@@ -0,0 +1,40 @@
+local ipairs = ipairs
+local re = require('ngx.re')
+
+local _M = {}
+
+local function last_caller_ip()
+  return ngx.var.remote_addr
+end
+
+local function ip_from_x_real_ip_header()
+  return ngx.req.get_headers()['X-Real-IP']
+end
+
+local function ip_from_x_forwarded_for_header()
+  local forwarded_for = ngx.req.get_headers()['X-Forwarded-For']
+
+  if not forwarded_for or forwarded_for == "" then
+    return nil
+  end
+
+  return re.split(forwarded_for, ',', 'oj')[1]
```

Comment:
This is more complicated than just taking the taking the first one.

The realip module http://nginx.org/en/docs/http/ngx_http_realip_module.html exposes some "internal" ranges, so they don't get considered to be client IPs.

I hope this is enough for the OpenShift case, as for example our echo-api (running on OpenShift) gets following headers:

```
X-Forwarded-For: 46.135.85.38, 10.0.103.238
Forwarded: for=10.0.103.238;host=echo-api.3scale.net;proto=http
```